core/include/cubos/core/data/ser/json.hpp

```diff
@@ -0,0 +1,31 @@
+/// @file
+/// @brief Class @ref cubos::core::data::JSONSerializer.
+/// @ingroup core-data-ser
+
+#pragma once
+
+/// [JSON package]
+#include <nlohmann/json.hpp>
+
+#include <cubos/core/data/ser/serializer.hpp>
+#include <cubos/core/memory/stream.hpp>
+
+namespace cubos::core::data
+{
+    /// @brief Implementation of the abstract Serializer class for serializing to JSON.
+    class JSONSerializer : public Serializer
+    {
+    public:
+        /// @brief Constructs.
```

Comment:
You can remove this constructor, in this case the default us enough.

Reply:
but there's no default constructor in Serializer?

Reply:
You don't have to define it, the compiler does it implicitly.

Reply:
ok, so to where do I move the AUTO_HOOK calls?

Reply:
I'm retarded, disregard kek